### ISSUE TITLE
0.62.x Discussion

### ISSUE THREAD

**Issue by grabbou** [react-native-community/releases#145]
Opening this issue thread as a placeholder that we will be starting to work on 0.62.0-rc.0 now that 0.61.0 is shipping.

I will post an update here once we have a better understanding of the timeline.

**Comment by grabbou** [comment]
Now that 0.61.2 has finally shipped fixing a long-standing elevation regression on Android, I am going to look into 0.62.

**Comment by ursNj** [comment]
What all major features planning for 0.62 ??? 

very much eager to know, that i loved the Fast refresh in 0.61 :)

Kudo's to the team.

**Comment by sebinq** [comment]
@ursNj Brand new react dev tools ❤️

**Comment by ursNj** [comment]
improving Flatlist performance is really needed, as many times i saw that application is lagging due to flat list. even i have tried all the optimization techniques for Flatlist, still issue is there. its is very much required.

**Comment by grabbou** [comment]
It is not specified yet. I will let you know as soon as the branch cut works and we are ready to push out a new release.

Right now, tests are failing because of two issues and I am waiting for an update.

**Comment by anchengjian** [comment]
Remove `create-react-class` and it's related writing to avoid duplication of `fbjs` dependency.

**Comment by grabbou** [comment]
The Android CI has been fixed. There's work to resolve iOS issues. I will update you as soon as I have more information.

**Comment by grabbou** [comment]
Update: I have reverted all commits that break the iOS tests. I am waiting for the CI to become green.

Here's a summary of reverted commits:

- https://github.com/facebook/react-native/commit/251fda207326c05c9c497530409a1f4e86f6ebfe - this commit changes gray shades in RNTester, causing our snapshot tests to fail. A fix should be to safely re-generate snapshots and make sure it's just the colors we are overwriting to ensure no bugs are slipping through. Alternatively, we can change colors in the theme to match previous RNTester colors

- https://github.com/facebook/react-native/commit/561cbbe2e9025f6f6ff3fb98267adb6502be99d8 - this removes Flipper integration for all iOS projects. Not sure why it breaks, but the failing test suite is [here](https://circleci.com/gh/facebook/react-native/117499) if you want to dig it yourself. The author have been notified.

- https://github.com/facebook/react-native/commit/0d54b1c23f9d8943cfa5380eb9a16cd4dd2ae34e - this reverts changes done to RCTBlob to make it compatible with TurboModules. Nothing that affects you right now, but it breaks the CI. The authors have been notified.

I don't have capacity to work on these right now. Please note that they are still present on `master`, making its CI red.

**Comment by grabbou** [comment]
Unfortunately, the tests are still failing because of the [TurboModules commits](https://github.com/facebook/react-native/commits?after=cb1171b37ae7e34fbcb608305d68f9e6ac601cde+34&author=RSNara) that are causing "Undefined symbols for architecture x86_64" error.

You can find a failure for `RCTSettings` module [here](https://circleci.com/gh/facebook/react-native/119525?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link).

I will try to work on resolving this issue and apply https://github.com/facebook/react-native/commit/0d54b1c23f9d8943cfa5380eb9a16cd4dd2ae34e again. I believe there's one fix needed for all modules that were made TurboModules-compatible to make compiler pass.

**Comment by grabbou** [comment]
[The tests are green](https://circleci.com/gh/facebook/react-native/120421?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link). I would like to thank everyone for the help with getting the commits and fixes in place! Especially @RSNara! 

We are now evaluating whether to release with October, 13th commits or to re-cut from master (will require few more days to fix tests).

**Comment by sebinq** [comment]
Re-cut 🥳😄

**Comment by kelset** [comment]
I agree - if CI is green on master I'd probably just re-cut it since it could at least reduce the difference between the released version and master from a month-ish to a few days / a week

**Comment by mikehardy** [comment]
"if CI is green on master"... - but, nope: master is solid red https://github.com/facebook/react-native/commits/master

Just for grins I kept paging back looking for a green check. July 18 is the last time master passed CI? https://github.com/facebook/react-native/commits/master?after=56c7ae729a03f44558763e3d86def3a269513bbb+1049

I'm not recommending a course of action but if I'm reading history right that's the facts

Actually, I would recommend getting CI to pass on master ;-) as a "come on everyone let's take a break for a minute and do this" type deal but I'm not sure I'd hold 0.62 for it

**Comment by kelset** [comment]
oh wow - yeah sounds like there should be an effort in getting master back to green that 👀

But then, if instead the current state of 0.62 is a green CI, I would proceed with the current one and not rebase.

**Comment by grabbou** [comment]
There's been a few fixes and tweaks here on 0.62-stable to make it pass and all of them have been backported to `master` since then. Unfortunately, that means we are dealing with a different failure reason now.

I'll check how time-consuming that might be and will report back.

**Comment by grabbou** [comment]
CC: @TheSavior and @cpojer, I'd love to get your thoughts on [wheter to release as is or sync with master](https://github.com/react-native-community/releases/issues/145#issuecomment-549440199). 

There's been some additional context in the recent comments too!

**Comment by cpojer** [comment]
I think the consideration should be about how many more releases we want to have this year and when. The two options I see are:

1. Stabilize 0.62 now and release it in November, cut a new stable at the end of the year and target a January release for 0.63
2. Re-cut 0.62 in mid-November and make a stable release in mid-December, then make 0.63 happen in February.

I don't want to influence your decision on this too much but given that you already put a lot of work into stabilizing 0.62, it's probably a good idea to release it (please write a brief blog post!) and then we'll cut the next release after the holidays.

**Comment by grabbou** [comment]
Thank you @cpojer for giving your input into the discussion. As you said, I think it's a good idea to release 0.62 right now, with the primary reason being the timing and avoiding unnecessary pressure throughout December.

I believe that releasing a stable version at the end of the year (say December) and cutting the next release candidate early next year (January) is the best way going forward and will help us wrap things up before getting offline for the New Year's Eve.

That said, I am going to proceed with it right now and will write a comment here once the results are online.

**Comment by grabbou** [comment]
There is a little issue with Android build system (similar to https://github.com/react-native-community/jsc-android-buildscripts/issues/80). Previously, it was only happening inside RNTester. Now, it also happens while building a new app.

Going to consult this with Facebook team and find a reasonable solution/workaround today before proceeding further.

Note: This shouldn't delay the release any longer, it's just a tiny little tweak I want to do before releasing to save everyone from doing the [fix](https://github.com/react-native-community/jsc-android-buildscripts/issues/80#issuecomment-462214364) themselves.

**Comment by thymikee** [comment]
CLI bump: https://github.com/facebook/react-native/commit/e523302fbe055e6745e38b72a17e121341c431c1

**Comment by grabbou** [comment]
Turns out the issue is brought up by recent Flipper update. Investigating with the team on the next steps.

**Comment by dnlowman** [comment]
Is there any update to the status of this?

**Comment by Fausto95** [comment]
is RN0.62 coming with Flipper support?

**Comment by grabbou** [comment]
@dnlowman yes - it took a while because I was awaiting an investigation from the Flipper team. There's a known workaround that is going to ship for now. We will see what to do with that next week.

@Fausto95 yes.

**Comment by grabbou** [comment]
I have already pushed a fix, going to test it and publish `0.62.0-rc.0`. 

It's been a while since I cut the branch, so I will take an additional time today & tomorrow to look at it once again and make sure everything works.

**Comment by ursNj** [comment]
app bundles crash issue in play store will fix with this release ???https://github.com/facebook/react-native/issues/25927

**Comment by grabbou** [comment]
Unfortunately, the Android app build is broken on my machine (despite the CI passing, where we don't test "a default app", but ReactAndroird as a whole). 

I need to take a closer look at the failures. 

In the meantime, let's focus on shipping additional tweaks to `0.61` - there's going to be more work on `0.62` needed to get it out. I will do my best to get it out as soon as possible, however, please keep in mind there is a lot of developers working on this behind the scenes and I am just delivering the updates.

That said, wish you a great weekend and hopefully, there's going to be more updates early next week.

**Comment by sebinq** [comment]
Hello @grabbou! Isn't it worth to consider re-cut and delay (release) once again?

**Comment by ltcaosj** [comment]
Can we include this PR to RN0.62? Allow touches on clipped children when `clipChildren==false` 
[PR #26374](https://github.com/facebook/react-native/pull/26374)
This will make a consistent UI behaviour between iOS and Android.

**Comment by DKolev** [comment]
Hello.

I was wondering if in any the upcoming releases, there is a fix related to the problematic cookie based authentication problem in Android. While we were on 0.59.9, we used the workaround with
`ReadableNativeArray.setUseNativeAccessor(true);`
 `ReadableNativeMap.setUseNativeAccessor(true);`

But we had to upgrade to the latest 0.61.4 and now our login for Android is completely broken.

**Comment by pinstripe-potatohead** [comment]
> Hello.
> 
> I was wondering if in any the upcoming releases, there is a fix related to the problematic cookie based authentication problem in Android. While we were on 0.59.9, we used the workaround with
> `ReadableNativeArray.setUseNativeAccessor(true);`
> `ReadableNativeMap.setUseNativeAccessor(true);`
> 
> But we had to upgrade to the latest 0.61.4 and now our login for Android is completely broken.

I believe this was fixed in [v0.61.5](https://github.com/facebook/react-native/releases/tag/v0.61.5)

**Comment by DKolev** [comment]
> > Hello.
> > I was wondering if in any the upcoming releases, there is a fix related to the problematic cookie based authentication problem in Android. While we were on 0.59.9, we used the workaround with
> > `ReadableNativeArray.setUseNativeAccessor(true);`
> > `ReadableNativeMap.setUseNativeAccessor(true);`
> > But we had to upgrade to the latest 0.61.4 and now our login for Android is completely broken.
> 
> I believe this was fixed in [v0.61.5](https://github.com/facebook/react-native/releases/tag/v0.61.5)

10x a lot. Indeed it was fixed. I guess 0.61.5 was released a couple of hours after my question here.

**Comment by akshetpandey** [comment]
I second holding off 0.62 release and recutting in January. The holiday season would mean low exposure for the rc builds and no one available to make fixes if required.

**Comment by kelset** [comment]
As pointed out by @cpojer a while ago, a lot of work went into stabilising 0.62 "as-is".

It looks like master branch CI is still red (like, 6 jobs out of 13 are failing 👀), and since that has been the case for a while (as pointed out by @mikehardy), I would probably instead go a different route.

@grabbou Android has always been a bit flaky during the local E2E testing - what if we just push out the 0.62-rc0 out as is and see what happens? I'm pretty sure that the `rc0` part of the release will keep most people "safe" from trying it, and we can get a larger pool of people to test it and detect what is currently failing on yours.

**Comment by grabbou** [comment]
@kelset unfortunately, the Android doesn't work right now. The tests are surprisingly passing, so it's not a matter of flaky tests this time.

What fails is the `react-native init`ed application, which is only tested manually and not on the CI. 

For reference, here's an excerpt from the stack traces:

#1 

```
2019-11-22 08:53:12.233 4932-4983/? E/AndroidRuntime: FATAL EXCEPTION: mqt_native_modules
    Process: com.rntestproject, PID: 4932
    java.lang.UnsatisfiedLinkError: No implementation found for long com.facebook.yoga.YogaNative.jni_YGConfigNew() (tried Java_com_facebook_yoga_YogaNative_jni_1YGConfigNew and Java_com_facebook_yoga_YogaNative_jni_1YGConfigNew__)
        at com.facebook.yoga.YogaNative.jni_YGConfigNew(Native Method)
        at com.facebook.yoga.YogaConfigJNIBase.<init>(YogaConfigJNIBase.java:23)
        at com.facebook.yoga.YogaConfigJNIFinalizer.<init>(YogaConfigJNIFinalizer.java:11)
       ...
```

#2 

```
2019-11-22 09:05:58.246 7133-7133/com.rntestproject E/SoLoader: Error when loading lib: dlopen failed: library "libjsc.so" not found lib hash: a0869b08725ad34d2f9c99602c9b316e search path is /data/app/com.rntestproject-tZ-YSeUocEWoe7yuj7iLBg==/lib/x86
2019-11-22 09:05:58.252 7133-7133/com.rntestproject E/SoLoader: couldn't find DSO to load: libjscexecutor.so caused by: dlopen failed: library "libjsc.so" not found
```

I am currently trying out a re-cut to see if the issues still persist there. It is easier to push with a broken CI (when some functionality might be broken) rather than with a not working app 😅

**Comment by kelset** [comment]
Oh, I see :( I misunderstood what you meant with Android not working 😓

Ok let us know how that goes 🤞

**Comment by cimitan** [comment]
Hi, did anyone try with git bisect to figure out the commit to blame for this regression?

**Comment by KingAmo** [comment]
hi, any update on `0.62`?

**Comment by grabbou** [comment]
I haven't tried that @cimitan and you're more than welcome to help us debug this! In the meantime, I have re-cut 0.62 branch off master under a different name and I am estimating the effort needed to unbreak the tests there.

The current 0.62-stable branch is blocked by the aforementioned Android issues which Flipper team was looking into fixing. No updates here.

**Comment by cimitan** [comment]
Quite busy this week, maybe I'll get the chance to test one of those nights. I'll report back if I start, shouldn't take long with bisection

**Comment by grabbou** [comment]
Facebook team removed all 0.62-stable branches after our request and there's now a new branch, straight off the master which seems to be working. 

We're doing tests right now and hoping to do the release pretty soon.

**Comment by robertying** [comment]
It seems 0.62 is close to release, but I’m wondering if there’s any chance to include this fix for iPad split views?

https://github.com/facebook/react-native/commit/7a72c35a20a18c19bf6ab883cb2c53a85bd4c5c0

Thank you!

**Comment by hramos** [comment]
@robertying I've cherry-picked 7a72c35a20a18c19bf6ab883cb2c53a85bd4c5c0 onto 0.62-stable.

**Comment by lukebars** [comment]
Any updates on estimated release date? :shipit:

**Comment by kelset** [comment]
At the moment, we are testing the new branch - 0.62.0 will happen in 2020. We may (or may not) manage to get an `rc` out before end of the year but no promises, sorry.

**Comment by gaodeng** [comment]
The release process of 0.62 has been going through almost three months 😱,Can we study Flutter's build release channels and learn from them . [https://github.com/flutter/flutter/wiki/Flutter-build-release-channels](url)

**Comment by kelset** [comment]
@gaodeng while I understand that you may find frustrating that it's been a slow process, I can assure you that we have been trying our best to make it happen in a reliable fashion (no one wants to release a broken version) - we have been documenting our efforts in this issue precisely so that we'd be transparent on our attempts. 

Also, I find a bit unfair for you to compare the two libraries as they have a really different workloads and approaches - as the releases for RN are mostly managed by members of the community, who are *not paid* to work on this.

It's already quite stressful to work on this process - without the additional negativity that comments like yours provide 😓 So please be more understanding of our position, there are humans on the other side of the screen.

**Comment by Off2Race** [comment]
@kelset — I hope you can ignore the negativity. There are many, many of us out here that greatly value the work that you and all the members of the community contribute. Keep up the great work. We appreciate you!

**Comment by gaodeng** [comment]
I just told the truth, I didn’t expect it would hurt you

**Comment by cimitan** [comment]
Issues and mistakes happen, it's human. Having worked for several years in the past in the free software community, I can assure those things happen, it's no big deal.

We need to learn and iterate, how can we improve the process for 0.63? That's going to be a separate discussion, not to be held here.

Thanks to the team for your work here, we appreciate the effort despite the busy holiday period!

**Comment by kelset** [comment]
Quick update on `0.62.0-rc.0`, since you may see the tag up in the main repo: I did some local testing of the `0.62-stable` branch and everything seemed to be working fine:
- RNTester Android :white_check_mark:
- RNTester iOS :white_check_mark:
- RNTestProject Android:white_check_mark:
- RNTestProject iOS:white_check_mark:

So today I attempted to proceed with the publication of the RC0 version - but this failed at the `publish_npm_package` [CircleCI job](https://circleci.com/gh/facebook/react-native/126321?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) because of a timeout:

```
The authenticity of host 'github.com (140.82.113.3)' can't be established.
RSA key fingerprint is SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8.
Are you sure you want to continue connecting (yes/no)? 
Too long with no output (exceeded 10m0s): context deadline exceeded
```

IIRC we had this issue in the past, but I think it needed some config update from someone from FB - I've communicated this to the core team and I'm waiting for a response (cc @hramos).

Hopefully this will be a non-code fix, meaning that fixing the config + re-running the job will be enough to send the `0.62.0-rc.0` version out.

I will write an update here tomorrow.

**Comment by kelset** [comment]
Update: `0.62.0-rc.0` is out on npm! 🚀

Turns out, I had the CircleCI permissions but I wasn't logged in 🤦‍♂️ I finally managed to log into the CircleCI vm via SSH and retrigger the command that was hanging (not the cleanest solution, but hey it worked).

I also already tested it locally, and at least on my machine a new fresh app created via
`npx react-native init RN062 --version next`

Worked cleanly:
<img width="300" alt="Screenshot 2019-12-18 at 17 01 47" src="https://user-images.githubusercontent.com/16104054/71107381-e2bda580-21b8-11ea-9ce5-06fc4d565a2e.png">

This is the part where you all come in 😉 if you want to help us out, it would be great if you could test this release (but please remember it's the *first ever release candidate* of 0.62) and report back here if you are finding any major issues or blockers.

If you find any, it would be great if you could also check the commit history on master for any commits that would be good candidates to fix the issue you saw!

At the moment, the shortlist of commits that I'm considering cherry-picking for RC1 (oldest-> newest):
* https://github.com/facebook/react-native/commit/68bf0e32e70db2ce28cc3c3d9d63f58900d1869d
* https://github.com/facebook/react-native/commit/d153e8a2099978f0ce72eeaf106c1171695ae132
* https://github.com/facebook/react-native/commit/69e9f3a3891a9b4a9acd58e97ae4e6f9ebcf968d

Just to clarify the timeline of the next steps: since the basic version is currently working, we will not do other releases this year. This means that `rc1` and subsequents versions (`rc2`, ..., `0.62.0`, etc) will all happen in 2020.

**Comment by a-c-sreedhar-reddy** [comment]
With which RN version is the latest React devtools compatible with?

**Comment by charpeni** [comment]
> @Fausto95: is RN0.62 coming with Flipper support?

Flipper is available on iOS in `0.62.0-rc.0`.

> @a-c-sreedhar-reddy: With which RN version is the latest React devtools compatible with?

The latest React devtools is compatible with `0.62.0-rc.0`, the same thing applies to Devtools inside Flipper.

**Comment by mjmasn** [comment]
Hey @kelset, sorry to be the bearer of bad news but I get this error when running `npx react-native init RN062 --version next` :thinking: 

```
Cannot find module '/home/mike/code/RN062/node_modules/react-native/package.json'
Require stack:
- /home/mike/.nvm/versions/node/v12.11.1/lib/node_modules/react-native-cli/index.js
```

and I end up with a directory just containing `node_modules`, `package.json` and `yarn.lock`

**Comment by mjmasn** [comment]
Maybe just a missing `next` tag? Works fine with `--version 0.62.0-rc.0`

**Comment by kelset** [comment]
thanks @mjmasn, I've updated the command!

**Comment by sofiageo** [comment]
Sorry for my ignorance, but will this rc be picked up by upgrade-helper? Does it require manual intervention or is it an automatic procedure that we will have to wait for? Thanks

**Comment by grabbou** [comment]
At this point, it has to be triggered manually. I am pretty sure somebody will do it early on.

Thank you @kelset for delivering the release - after few months of hard work and re-cutting the branch, we managed to ship 0.62.0-rc.0 this decade (:joy:).

For context: I am on holidays till end of the year. I will resume with processing your feedback next decade.

Wish you great Christmas time and a Happy New Year!

**Comment by tomtargosz** [comment]
Thanks to everyone who has been working on this, especially during such a hectic time of year! Can't stand seeing negativity in here when you're all dedicating your own time to help the community out. You all rock!

**Comment by pvinis** [comment]
Awesome, congrats!

I'll start 😄 😬 

I generated the diffs for purge and helper, and I noticed that `xcshareddata` is in gitignore, which then causes `ios/RnDiffApp.xcodeproj/xcshareddata/xcschemes/RnDiffApp.xcscheme` to be removed, when they should be there. They are the shared schemes so xcode knows what and how to build, on everyones machine. So I made https://github.com/facebook/react-native/pull/27563

**Comment by pvinis** [comment]
@sofiageo 

> Sorry for my ignorance, but will this rc be picked up by upgrade-helper? Does it require manual intervention or is it an automatic procedure that we will have to wait for? Thanks

It's manual, and I did it a few minutes ago. It's available. While upgrading using that, please ignore the `.gitignore` change as well as the two Xcode schemes that are deleted. Don't delete them. I submitted the PR above to fix this on rc.1 I guess.

**Comment by rickhanlonii** [comment]
Just created an issue to track the Flipper support, completing these items should be considered a blocker for 0.62: https://github.com/facebook/react-native/issues/27565

**Comment by rickhanlonii** [comment]
I just cherry-picked 6 commits to 0.62 for fixes to LogBox:

<img width="999" alt="Screen Shot 2019-12-19 at 10 34 52 AM" src="https://user-images.githubusercontent.com/2440089/71186483-4b556280-224b-11ea-8542-fb66c6fc9b31.png">

**Comment by beetlebum** [comment]
Not sure if this is the right place to post it, but I couldn't get Flipper to work on iOS.

I noticed that upgrade helper didn't include the call to `[self initializeFlipper:application];` in `didFinishLaunchingWithOptions`, so adding it fixes the issue.
Other than that, I've upgraded my app with only a minor issue in ART that was fixed last night.

Thanks for the hard work.

**Comment by ltcaosj** [comment]
Can we cherry pick this PR: https://github.com/facebook/react-native/pull/26374
"Android: Allow touches on clipped children when `clipChildren==false`"
We can create a complex and fancy UI with this support. Currently it's working fine on iOS but Android.

**Comment by charpeni** [comment]
@beetlebum 

> Not sure if this is the right place to post it, but I couldn't get Flipper to work on iOS.
> 
> I noticed that upgrade helper didn't include the call to `[self initializeFlipper:application];` in `didFinishLaunchingWithOptions`, so adding it fixes the issue.
> Other than that, I've upgraded my app with only a minor issue in ART that was fixed last night.

Correct. The setup for Flipper has been added to 0.62 for iOS and Android, but in order to get it working, it has to be enabled.

For Android, `add initializeFlipper(this, getReactNativeHost().getReactInstanceManager());` to `MainApplication.java`'s onCreate method.

For iOS, add `[AppDelegate initializeFlipper:application];` as the first line in `AppDelegate.m`'s `didFinishLaunchingWithOptions` method.

Once Flipper setup will be finalized, it will be added to the React Native website.

**Comment by safaiyeh** [comment]
So just to clarify, Flipper will not be initialized by default? Developers who start a fresh RN app with 0.62 have to add `initializeFlipper(this, getReactNativeHost().getReactInstanceManager());` and `[AppDelegate initializeFlipper:application];` to enable Flipper?

Which is a change from 0.61 where Flipper was initialized in the Android app?

**Comment by charpeni** [comment]
@safaiyeh this is correct.

I do not have the context on why Flipper wasn't meant to be enabled by default, maybe @axe-fb can provide more information on this.

**Comment by axe-fb** [comment]
We did mean to enable Flipper by default on both Android and iOS in debug mode, but there were CI failures due to which we had to revert my changes. I think for 0.62, @rickhanlonii  is working on ensuring that it is turned back on ?

**Comment by charpeni** [comment]
Sounds good. I was confused by the last part of your blog post saying that Flipper is by default, but we need to initialize it.

>Hence, for versions of React Native >= 0.62
>
>For Android, simply add initializeFlipper(this, getReactNativeHost().getReactInstanceManager()); to MainApplication.java's onCreate method.
>
>For iOS, add [AppDelegate initializeFlipper:application]; as the first line in AppDelegate.m's didFinishLaunchingWithOptions method.

But it seems like they're no reason to not initialize this by default then since the CI has been fixed. The remaining work is tracked in https://github.com/facebook/react-native/issues/27565.

Thanks for the context!

**Comment by safaiyeh** [comment]
I had a PR that I closed before I got context adding it to iOS.

I could push the Android change to it.
https://github.com/facebook/react-native/pull/27569

**Comment by sunnylqm** [comment]
Please cherry pick https://github.com/facebook/react-native/commit/5ddf00ee1acbf66c7204227c398a58c13e4545cf

**Comment by rickhanlonii** [comment]
@sunnylqm done

**Comment by terrysahaidak** [comment]
Any chance this will got cherry picked once landed to master?
https://github.com/facebook/react-native/pull/27506

I know this one is not a bug fix, but it's super awesome thing to have ASAP.

I've already done cherry picks in 0.60 and 0.61 branches in my fork of RN, so it should be pretty easy. I can help to do it if so :)

**Comment by waqas19921** [comment]
Any chance to have this functionaliy
https://github.com/facebook/react/pull/13211
I know this one is not a bug fix, and the pull request is too old. Can we add this type of functionality?

**Comment by danielgindi** [comment]
Can you add this #27607, as in iOS 13 there are serious bugs with `KeyboardAvoidingView`

**Comment by safaiyeh** [comment]
Could we get: https://github.com/facebook/react-native/pull/27512 cherry picked?

We have an open issue in react-native-cookie-store that requires use of this function. Without it, we cannot set Cookies on Android making it inconsistent with the iOS functionality.

https://github.com/safaiyeh/react-native-cookie-store/issues/1

**Comment by huuthinh245** [comment]
Pls get facebook/react-native#946 Have issue with EventEmitter

**Comment by feruzm** [comment]
Could this be also added? Android animation bug https://github.com/facebook/react-native/issues/21801

**Comment by kelset** [comment]
folks, can I please remember you all to only point to commits already merged on master?

Sometimes PR take time to be merged and things change in the meantime 😓

@huuthinh245 you are linking to a PR for another library, so I'm not sure what could be done on this side.

**Comment by xiao99xiao** [comment]
Could we get https://github.com/facebook/react-native/commit/f21fa4ecb73551bdc4c3d70db9fc13e93b19b3a6 cherry picked? Fixing the RCTWebsocket means a lot to Catalyst developers.

**Comment by mgcrea** [comment]
Failed to build a clean project using this as I'm encountering the following error:

```
[...]/ios/Pods/Headers/Private/Flipper-Folly/folly/portability/Time.h:51:17: Typedef redefinition with different types ('uint8_t' (aka 'unsigned char') vs 'enum clockid_t')
```

Any ideas?

Aslo, using `--template react-native-template-typescript` won't work (I guess might be normal as it is a rc).

**Comment by kelset** [comment]
@mgcrea AFAIK the FB team is going to cherry pick a few more Flipper-related commits/bugfixes before next RC, so hopefully in next it will be solved! Thanks for reporting anyway :)

**Comment by alloy** [comment]
Assuming it lands, [this](https://github.com/facebook/react-native/pull/27731) needs to be picked too. That PR is a version of https://github.com/facebook/react-native/pull/27730 that’s already updated for the 0.62 branch.

**Comment by art1373** [comment]
I hope the you fixed how toast is not rendered on modals this is a huge issue. please customize the modal to let toast render on top of them ( custom toast I mean)

**Comment by linjson** [comment]
@grabbou  When do you enable SafeAreaView on Android?

**Comment by vanhungoz** [comment]
could this release fix random crash in production in this issue [https://github.com/facebook/react-native/issues/27463](https://github.com/facebook/react-native/issues/27463)

**Comment by RikScheffer** [comment]
is it possible to cherry pick https://github.com/facebook/react-native/commit/1fbc6a7c178d13421b0b84d6ea01f9174105325f ?

**Comment by surajSDM** [comment]
@grabbou you need to change pod file dependency i.e   pod 'ReactCommon/jscallinvoker', :path => "../node_modules/react-native/ReactCommon" 
to   pod 'ReactCommon/callinvoker', :path => "../node_modules/react-native/ReactCommon"
as this is defined inside ReactCommon.podspec

**Comment by jacobp100** [comment]
Congrats on the rc1 release! 😄

I noticed the changes to this vs the rc0 were pretty small - is that usually the case for RCs? I'm asking because I was just going to hold off upgrading until it's fully out, but if the changes are this small, I might get a head start on that upgrade

Thanks again for your work!

**Comment by kelset** [comment]
Hey everyone - welcome to 2020! 

We just published a new RC that introduces the possibility to test the new LogBox, you can find the release notes here: https://github.com/facebook/react-native/releases/tag/v0.62.0-rc.1

In an effort to de-noise this issue which has been open since Sept 2019 let's move the conversation for this Rc phase to a new issue -> https://github.com/react-native-community/releases/issues/157

---

@jacobp100 the number of commits was quite small because during the holidays we were considering also adding another number of changes to enable Flipper. But since it's going to take a bit before those lands (and also the FB team is investigating some Android-Hermes issues) we decided to do a first RC now to enable everyone to try the LogBox. And give some more breathing room for Flipper & Hermes to land.

You should wait until 0.62.0 to upgrade, as we are planning to do at least one more RC introducing some more things. We will communicate when we are publishing "the last RC" before .0. Overall, the process will be as fast as we can handle it and all the help and feedback from developers from the community we can get will help us!
